04.ls/ls.rb

```diff
@@ -20,6 +49,70 @@ def prepare_files(options)
   options['r'] ? sorted_files.reverse : sorted_files
 end
 
+def handle_long_format(files)
+  file_details = []
+  total_512_blocks = 0
+
+  files.each do |filename|
+    filestat = File.stat(filename)
+    total_512_blocks += filestat.blocks
+    file_details << organize_file_details(filename, filestat)
+  end
+  output_long_format(total_512_blocks, file_details)
+end
+
+def organize_file_details(filename, filestat)
+  file_permissions_string = organize_file_permission_string(filestat)
```

Comment:
```suggestion
  file_permissions = build_file_permissions(filestat)
```

organize=整理するというより、文字列の構築=buildの方がいいかも。
あと、`_string` もちょっと冗長なので省略してもいいかも。

Reply:
メソッド名を複数見直しました🙇‍♀️
- `handle_long_format`→`prepare_file_details`
元々出力メソッドも`handle_long_format`に含めていたのですが、切り離したため、ブロック数とファイル詳細の配列を準備するということで`prepare_file_details`にしました。
`build_file_details`もいいかなと思ったのですが、その下のメソッドにもbuildを使ってしまったので、区別しやすいようprepareとしてみました。
- `organize_file_details` →　`build_file_detail`
ファイル詳細を入れる配列がfile_detailsなので、このメソッド自体は単数にした方がいいかと思い、`build_file_detail`にしました。
- `_string`を削除しました。